### ISSUE TITLE
[swift] CPD  `#unavailable(_)` breaks token parsing

### ISSUE THREAD

**Issue by msworking** [pmd/pmd#4186]
**Affects PMD Version:** 
Unknown, latest version via `brew update pmd` as of 31 Oct 2022 (`pmd -v` gives `-v is NOT a valid application name`). Was affecting a previous version going back four months or better.

Make sure, to test with the latest PMD version.

**Description:**

The `#unavailable(_)` method in Swift generates an unrecognized token error. (Meanwhile the `#available(_, _)` method causes no issues.)

**Exception Stacktrace:**

```
net.sourceforge.pmd.lang.ast.TokenMgrError: Lexical error in file /Users/mswan/Documents/Code/kaz_ios_ALT/./learningazquizroom/LAZBarUIUtility.swift at line 80, column 14.  Encountered: token recognition error at: '#u'
	at net.sourceforge.pmd.cpd.AntlrTokenizer.tokenize(AntlrTokenizer.java:44)
	at net.sourceforge.pmd.cpd.CPD.addAndThrowLexicalError(CPD.java:154)
	at net.sourceforge.pmd.cpd.CPD.add(CPD.java:148)
	at net.sourceforge.pmd.cpd.CPD.add(CPD.java:117)
	at net.sourceforge.pmd.cpd.CPD.add(CPD.java:79)
	at net.sourceforge.pmd.cpd.CPD.addDirectory(CPD.java:90)
	at net.sourceforge.pmd.cpd.CPD.addRecursively(CPD.java:74)
	at net.sourceforge.pmd.cpd.CPDCommandLineInterface.addSourcesFilesToCPD(CPDCommandLineInterface.java:162)
	at net.sourceforge.pmd.cpd.CPDCommandLineInterface.addSourceFilesToCPD(CPDCommandLineInterface.java:142)
	at net.sourceforge.pmd.cpd.CPD.runCpd(CPD.java:225)
	at net.sourceforge.pmd.cpd.CPD.main(CPD.java:196)
Caused by: LexerNoViableAltException('#')
	at org.antlr.v4.runtime.atn.LexerATNSimulator.failOrAccept(LexerATNSimulator.java:312)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.execATN(LexerATNSimulator.java:233)
	at org.antlr.v4.runtime.atn.LexerATNSimulator.match(LexerATNSimulator.java:117)
	at org.antlr.v4.runtime.Lexer.nextToken(Lexer.java:141)
	at net.sourceforge.pmd.lang.antlr.AntlrTokenManager.getNextTokenFromAnyChannel(AntlrTokenManager.java:47)
	at net.sourceforge.pmd.lang.antlr.AntlrTokenManager.getNextToken(AntlrTokenManager.java:40)
	at net.sourceforge.pmd.cpd.token.internal.BaseTokenFilter.getNextToken(BaseTokenFilter.java:45)
	at net.sourceforge.pmd.cpd.AntlrTokenizer.tokenize(AntlrTokenizer.java:39)
	... 10 more
Oct 31, 2022 4:15:07 PM net.sourceforge.pmd.cpd.CPD runCpd
SEVERE: An error occurred while executing cpd.
```

**Code Sample demonstrating the issue:**

```swift
import UIKit

class LAZBarUIUtility {
    …
    
    static func setBarColors(navBar: UINavigationBar?, tabBar: UITabBar?) {
        guard #unavailable(iOS 13) else { return }
        …
    }
}

```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  In Terminal navigate to the root level of a Swift project
2.  Run ` pmd cpd --files . --minimum-tokens 50 --language swift --encoding UTF-8 --format net.sourceforge.pmd.cpd.XMLRenderer > cpd-output.xml`

**Running PMD through:** CLI (Terminal) and Xcode (Run script build phase)

**Workaround**
Use the `--exclude` flag to exclude the offending file(s). This allows cpd to run on the rest of the code.

Example: Using the following on the command line allowed for most of the project code to be scanned in this instance: ` pmd cpd --files . --minimum-tokens 50 --language swift --encoding UTF-8 --format net.sourceforge.pmd.cpd.XMLRenderer > cpd-output.xml --exclude ./learningazquizroom/LAZBarUIUtility.swift`

**Comment by jsotuyod** [comment]
Thanks for reporting this. Effectively, the Swift grammar currently in use is a little dated (supporting up to Swift 5.2), so features such as `#unavailable` (added in Swift 5.6) may fail to parse.

Unfortunately, the [original grammar we used](https://github.com/sleekbyte/tailor/blob/master/src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4) seems to be unmaintained and the [official Swift grammar from Antlr](https://github.com/antlr/grammars-v4/tree/master/swift/swift5) was last updated to Swift 5.4.

Updating the grammar is feasible. Supporting `#unavailable` is probably trivial on its own, but not good enough to properly claim support for newer Swift versions. Can you provide a link to an exhaustive changelog since Swift 5.4?

**Comment by jsotuyod** [comment]
This was fixed with #4384 and part of PMD 6.55.0 and later.
